sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -285,6 +285,17 @@ private void finalizeTransaction(final @NotNull IScope scope, final @NotNull Hin
 
     event = processReplayEvent(event, hint, options.getEventProcessors());
 
+    if (event != null) {
```

Comment:
I'd move this block after the "dropped by event processor" log (L302) to be more readable imho
Also, as it is now, if the replay is dropped by beforeSendReplay, we log `Event was dropped by beforeSendReplay` and right after that `Replay was dropped by Event processors.` (it's missing `return SentryId.EMPTY_ID;`)

Reply:
Ok, I just removed the log line for now, because we're already logging it in `processReplayEvent` anyway, so this log was excessive. I'll keep the order as-is, as it reflects what we have in `captureEvent` as well